rts/Map/ReadMap.cpp

```diff
@@ -559,13 +559,64 @@ void CReadMap::UpdateHeightBounds(int syncFrame)
 		tempHeightBounds.x = std::numeric_limits<float>::max();
 		tempHeightBounds.y = std::numeric_limits<float>::lowest();
 	}
-
+	
+#if 0
 	for (int idx = idxBeg; idx < idxEnd; ++idx) {
 		float h = (*heightMapSyncedPtr)[idx];
 
 		tempHeightBounds.x = std::min(h, tempHeightBounds.x);
 		tempHeightBounds.y = std::max(h, tempHeightBounds.y);
 	}
+#else
+	// // Cache align check
+	// int alignOff = ((int)(*heightMapSyncedPtr)[0]) & 0x3f; // MSBs are not needed
+
+	// if (alignOff != 0) { // well shit...
+	// // TODO sort this out
+	// }
+
+	int elements = idxEnd - idxBeg;
+	static const int chunkSize = 16; // 64 / sizeof(float)
+	int minChunksForCacheLine = elements / chunkSize;
+	minChunksForCacheLine = elements % chunkSize ? minChunksForCacheLine + 1 : minChunksForCacheLine;
+
+	int maxThreads = ThreadPool::GetMaxThreads();
+	int chunksPerThread = 1;
+	int usingThreads = minChunksForCacheLine;
+
+	if (minChunksForCacheLine > maxThreads) {
+		chunksPerThread = minChunksForCacheLine / maxThreads;
+		chunksPerThread = minChunksForCacheLine % maxThreads ? chunksPerThread + 1: chunksPerThread;
+		usingThreads = maxThreads;
+	}
+
+	std::vector<float2> threadResults(usingThreads);
```

Comment:
should be `std::array<float2, ThreadPool::MAX_THREADS>`, this can easily live on the stack